web/server/codechecker_server/api/report_server.py

```diff
@@ -1863,6 +1870,63 @@ def getCheckerCounts(self, run_ids, report_filter, cmp_data, limit,
                 results.append(checker_count)
         return results
 
+    @exc_to_thrift_reqfail
+    @timeit
+    def getAnalyzerNameCounts(self, run_ids, report_filter, cmp_data, limit,
+                              offset):
+        """
+          If the run id list is empty the metrics will be counted
+          for all of the runs and in compare mode all of the runs
+          will be used as a baseline excluding the runs in compare data.
+        """
+        self.__require_access()
+
+        limit = verify_limit_range(limit)
+
+        results = {}
```

Comment:
Shouldn't this contain analyzers as keys and 0 as value? Does it work if empty dict is returned?

Reply:
Yes, it will work, for example we do the same thing in `getSeverityCounts` API function.

For example if the database is empty, so there isn't any report in the database then this function will return an empty dictionary.

This function returns a dictionary where the keys are analyzer names which can be found in the database.